### PR TITLE
Add AWS secrets Manager plugin.

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: pkg/plugins/secret/secrets_manager/secrets_manager_test.go [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+				})
+			})
+
+			When("AWS SecretsManager.GetSecretValue returns an non NOT_FOUND error", func() {
+				ctrl := gomock.NewController(GinkgoT())
+				mockSecretClient := mocks.NewMockSecretsManagerAPI(ctrl)
+				secretPlugin := &secretsManagerSecretService{
+					client: mockSecretClient,
+				}
+				It("Should successfully store a secret", func() {
```
Comment (medgar-nitric) [inline pkg/plugins/secret/secrets_manager/secrets_manager_test.go:148]: Change `It("..", func() {` to reflect scenario

File: pkg/plugins/secret/secrets_manager/secrets_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+				})
+			})
+
+			When("AWS SecretsManager.PutSecretValue returns an AWS error", func() {
+				ctrl := gomock.NewController(GinkgoT())
+				mockSecretClient := mocks.NewMockSecretsManagerAPI(ctrl)
+				secretPlugin := &secretsManagerSecretService{
+					client: mockSecretClient,
+				}
+				It("Should successfully store a secret", func() {
```
Comment (medgar-nitric) [inline pkg/plugins/secret/secrets_manager/secrets_manager_test.go:170]: Change `It("..", func() {` message to reflect scenario

File: pkg/plugins/secret/secrets_manager/secrets_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+					).Return(nil, fmt.Errorf("failed to get secret")).Times(1)
+
+					response, err := secretPlugin.Access(&secret.SecretVersion{
+						Secret: &secret.Secret{
+							Name: "test-id",
+						},
+						Version: "test-version-id",
+					})
+					By("Not returning an error")
+					Expect(err).Should(HaveOccurred())
```
Comment (medgar-nitric) [inline pkg/plugins/secret/secrets_manager/secrets_manager_test.go:248]: We probably need to document this behavior in the SDK's that an exception will be thrown secret not found.

**Review by medgar-nitric — CHANGES_REQUESTED** [review]

File: pkg/plugins/secret/secrets_manager/secrets_manager.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+func (s *secretsManagerSecretService) validateNewSecret(sec *secret.Secret, val []byte) error {
+	if sec == nil {
+		return plugins.NewInvalidArgError("provide non-empty secret")
+	}
+	if len(sec.Name) == 0 {
+		return plugins.NewInvalidArgError("provide non-blank secret name")
+	}
+	if len(val) == 0 {
+		return plugins.NewInvalidArgError("provide non-blank secret value")
```
[the remark this answers is not in the document]
Comment (tjholm) [reply pkg/plugins/secret/secrets_manager/secrets_manager.go]: done

File: pkg/plugins/secret/secrets_manager/secrets_manager.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+	}
+}
+
+func (s *secretsManagerSecretService) Access(sv *secret.SecretVersion) (*secret.SecretAccessResponse, error) {
+	if len(sv.Secret.Name) == 0 {
+		return nil, fmt.Errorf("provide non-blank secret name")
+	}
+	if len(sv.Version) == 0 {
+		return nil, fmt.Errorf("provide non-blank version")
+	}
```
[the remark this answers is not in the document]
Comment (tjholm) [reply pkg/plugins/secret/secrets_manager/secrets_manager.go:121]: done

File: pkg/plugins/secret/secrets_manager/secrets_manager_test.go (hunk 2 above)
Comment (tjholm) [reply pkg/plugins/secret/secrets_manager/secrets_manager_test.go]: done

File: pkg/plugins/secret/secrets_manager/secrets_manager_test.go (hunk 3 above)
Comment (tjholm) [reply pkg/plugins/secret/secrets_manager/secrets_manager_test.go:248]: done

**Review by medgar-nitric — APPROVED** [review]

**Review by jyecusch — APPROVED** [review]

**Review by jyecusch — APPROVED** [review]

**Review by medgar-nitric — APPROVED** [review]

**Pull request merged**